### DISCUSSION TITLE
Cdg

### DISCUSSION THREAD

**Discussion by karl-micesoft** [nschlia/ffmpegfs#147] [Q&A]
Hi all

I have a library of karaoke in MP3+CDG format (i.e. an MP3 file for audio, plus a binary CDG file containing the karaoke graphics). Ffmpeg is able to convert these to a real video format fairly easily, but they take up significantly more space.

So, my question: will ffmpegfs support these in the source directory? I'll give it a try myself when I get a chance, but asking in advance so I don't wake my time if someone already knows it won't.

Thanks in advance

**Comment by nschlia** [comment]
If the file that is created out of the CDG is too large it should be possible to get it smaller by playing with the target compression rates.

I could give it a try, do you have an example MP3+CDG you could provide me with (of course no copyrighted material).

But why is the size a problem? It's a virtual file anyway, or do you want to copy the converted files somewhere?

**Comment by karl-micesoft** [comment]
So, I may not have been clear. I would like to be able to use `ffmpegfs`, because converting all my karaoke tracks would take up a lot more space. I don't mind the size of the converted file if it's not being permanently stored.

If you could test it in advance, that would be brilliant. The following page has sample files available to download for testing purposes: https://zoom-entertainments.co.uk/free-samples

**Reply by nschlia** [reply]
> So, I may not have been clear. I would like to be able to use `ffmpegfs`, because converting all my karaoke tracks would take up a lot more space. I don't mind the size of the converted file if it's not being permanently stored.

Now I 've got it. You meant you can convert with **ffmpeg**, not **ffmpegfs**. I just tried what happens when I convert a MP3/CDG tandem to MP4 with ffmpegfs. What I get is two separate MP4 files which is relatively useless. The video that was created also looked very ugly as well...  

I guess what you had in mind is to get a combined MP4 or whatever with the video and the original (not recoded) MP3 audio (to preserve the quality).

I can add that feature, sounds quite handy, but please be patient. I am quite busy ATM

**Reply by karl-micesoft** [reply]
Yes, that's what I'm after.

If you could do this, it would be amazing. I'm happy to wait, and to test as you need it. I'm a dev myself and know how much time and effort such activities can take.

**Reply by karl-micesoft** [reply]
In fact, I think I could see another use case for similar functionality: Combining Tesla DashCam footage into a single file with multiple video tracks.

**Reply by nschlia** [reply]
> Yes, that's what I'm after.
> 
> If you could do this, it would be amazing. I'm happy to wait, and to test as you need it. I'm a dev myself and know how much time and effort such activities can take.

Finally I got more time now, but first I want to prepare another release. After that I'll get into #148 and add the CDG feature. As a matter of fact I can use it myself (for singing exercises), so I am interested in it.

**Reply by nschlia** [reply]
> In fact, I think I could see another use case for similar functionality: Combining Tesla DashCam footage into a single file with multiple video tracks.

Please create an issue for that. If possible, add footage examples or links to some. Also, if there's a specification by Tesla, it would be useful as well.
